### PR TITLE
Show component location for selected element in bottom/right of props panel

### LINKED ISSUE #16497: DevTools: Show component file path
**Do you want to request a *feature* or report a *bug*?**

Feature.

**What is the current behavior?**

The new DevTools doesn't show the component file path as did the v3 version.

**What is the expected behavior?**

It would be great to show the component file path, it's a really important feature when working in a large codebase.

**Which versions of React, and which browser / OS are affected by this issue? Did this work in previous versions of React?**

React DevTools v4 is affected, v3 had it.

### PR DESCRIPTION
**Pull request by bvaughn**
Resolves #16497

Screen Shot 2019-12-10 at 8 46 02 AM

Rational for placement and formatting here:
https://github.com/facebook/react/issues/16497#issuecomment-564122209

### CODE REVIEW

File: packages/react-devtools-shared/src/devtools/views/Components/SelectedElement.js [hunk 1]
```diff
@@ -403,6 +405,51 @@ function InspectedElementView({
...
+
+      {source !== null && (
+        <Source fileName={source.fileName} lineNumber={source.lineNumber} />
+      )}
+    </div>
+  );
+}
+
+// The function below (formatSourceForDisplay) is based on packages/shared/describeComponentFrame.js
+const BEFORE_SLASH_RE = /^(.*)[\\\/]/;
```
Comment (threepointone) [inline packages/react-devtools-shared/src/devtools/views/Components/SelectedElement.js]: Could you move this inside formatSourceForDisplay? Regexes are mutable, and I'm always wary of sharing them across function runs.

Comment (bvaughn) [reply packages/react-devtools-shared/src/devtools/views/Components/SelectedElement.js]: Sure. This follows the precedent in other files (like `describeComponentFrame`) but I don't mind making the change here.

**Comment by gaearon** [comment]
Yay!

**Comment by jschuler** [comment]
Is this feature still supported? I don't see the Source on selected elements

**Pull request merged**